src/main/java/com/redhat/microprofile/psi/internal/quarkus/core/properties/QuarkusConfigMappingProvider.java

```diff
@@ -201,6 +202,8 @@ private void populateConfigObject(PsiClass configMappingType, String prefixStr,
 						resolvedTypeSignature = getRawResolvedTypeName(((PsiClassType) psiType).getParameters()[0]);
 						returnType = findType(method.getManager(), resolvedTypeSignature);
 						simpleType = isSimpleType(resolvedTypeSignature, returnType);
+					} else if (isInJavaPackage(returnType)) {
+						simpleType = true;
```

Comment:
Could you explain which isecase you are trying to cover? java.utl.List is not a sipleType if I remember.

Reply:
`java.util.List ` is a collection and is thus catched by the first check. This is needed so that `OptionalInt`, `OptionalLong` etc. wont make it into the recursion at line 217

Reply:
changed it, the recursion is now avoided for simple types and types in the java package.